File: Kattis/Java/Relocation.java
package relocation;

import java.util.Scanner;

public class Relocation {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int numOfCompanies = input.nextInt();
        int numOfRequests = input.nextInt();
        int[] initialLocations = new int[numOfCompanies];
        
        for (int i = 0; i < numOfCompanies; i++) {
            initialLocations[i] = input.nextInt();
        }
        for (int j = 0; j < numOfRequests; j++) {
            int action = input.nextInt();
            int company = input.nextInt();
            int location = input.nextInt();
            
            if (action == 1) {
                initialLocations[company - 1] = location;
            } else {
                System.out.println((Math.abs(initialLocations[company - 1] - 
                        initialLocations[location - 1])));
            }
        }
    }
    
}

File: Kattis/CPlusPlus/Relocation.cpp
#include <iostream>
#include <vector>

int main() {
	int numOfCompanies, numOfRequests, action, company, location;

	std::cin >> numOfCompanies >> numOfRequests;
	std::vector<int> initialLocations(numOfCompanies);
	
	for (int i = 0; i < numOfCompanies; i++) {
		std::cin >> initialLocations[i];
	}
	for (int i = 0; i < numOfRequests; i++) {
		std::cin >> action >> company >> location;
		
		if (action == 1) {
			initialLocations[company - 1] = location;
		}
		else {
			std::cout << (abs(initialLocations[company - 1] - initialLocations[location - 1]));
		}
	}

	system("PAUSE");
	return 0;
}